tests/test_reproducibility.py

```diff
@@ -36,7 +35,39 @@ def test_voice_reproducibility(self):
 
     def test_voice_nonreproducibility(self):
         with pytest.raises(ValueError):
-            SynthConfig(batch_size=BATCH_SIZE_FOR_REPRODUCIBILITY + 1)
+            SynthConfig(batch_size=BASE_REPRODUCIBLE_BATCH_SIZE + 1)
+
+    def test_voice_batch_size_reproducibility(self):
+        # Test to make sure different batch sizes produce the same results
+        voice256 = Voice(SynthConfig(batch_size=256))
+        out256 = voice256(0)
+
+        voice128 = Voice(SynthConfig(batch_size=128))
+        out128 = torch.vstack([voice128(0), voice128(1)])
+
+        voice64 = Voice(SynthConfig(batch_size=64))
+        out64 = torch.vstack([voice64(0), voice64(1), voice64(2), voice64(3)])
+
+        voice32 = Voice(SynthConfig(batch_size=32))
+        out32 = torch.vstack(
+            [
+                voice32(0),
+                voice32(1),
+                voice32(2),
+                voice32(3),
+                voice32(4),
+                voice32(5),
+                voice32(6),
+                voice32(7),
+            ]
+        )
+
+        # TODO there are some unexpected, very small numerical
+        # errors between some, but not all, batch sizes
+        # See https://github.com/torchsynth/torchsynth/issues/326
+        assert torch.all(torch.isclose(out256, out128))
+        assert torch.all(torch.isclose(out256, out64))
+        assert torch.all(torch.isclose(out256, out32))

```

Comment:
This was weird. I was initially checking to see if they were equal and out256 == out64 and out256 == out32, but, out256 != out128. Looked into it and there are 40 samples all with an error of 7.450581e-09 at about 3min 40s into synth1B1-182. But the error only exists between batch size of 128 and the others. I checked the parameters and the noise and they are exactly the same. This one is boggling me a bit lol.

Reply:
On GPUs? They're not deterministic at low epsilons. The comment is good

Reply:
Ya I'll extend this test to check on GPUs if they're available